src/main/java/org/plumelib/bcelutil/StackMapUtils.java

```diff
@@ -379,16 +399,16 @@ protected final void modify_stack_maps_for_switches(InstructionHandle ih, Instru
   }
 
   /**
-   * Find the live range of the compiler temp(s) at the given offset and create a LocalVariableGen
-   * for each. Note the compiler might generate temps of different sizes at the same offset (must
-   * have disjoint lifetimes).
+   * Find the live range of the compiler temp(s) and/or user declared locals at the given offset and
+   * create a LocalVariableGen for each. Note the compiler might generate temps of different sizes
+   * at the same offset (must have disjoint lifetimes).
    *
    * @param mgen the method
-   * @param offset compiler assigned local offset of hidden temp
-   * @return offset incremented by size of smallest temp found at offset
+   * @param offset compiler assigned local offset of hidden temp or local
+   * @return offset incremented by size of smallest variable found at offset
```

Comment:
There can be multiple variables at the same offset; we need to know the size of the smallest to properly increment the offset when fix_local_variable_table loops through all local offsets looking for 'missing' locals.

Reply:
My question is about the meaning of "properly increment".

Suppose that the compiler generates two local variables (with different live ranges) at the same offset.  One is a bool and the other is a double, so their sizes are different.  This method returns an offset that is beyond the bool, but is not beyond the double.

If some code depends on the fact that the returned value is a valid place to put a different local variable v2, there might be an overlap between v2 and the double: both could refer to some of the same bytes.  (There would be no overlap in the local variable table, though, since it contains only enough information for the bool.)

Is this a problem?

That's why I thought that the max offset might be more appropriate, to avoid any overlapping variables.

Maybe this is not relevant, and all that matters is that there is *some* local variable table information about each variable, whether or not it is accurate.

Or maybe I am confused and the computed local variable table information is accurate.

Reply:
I have assumed that javac does not make mistakes.  If it allocates a bool and then an int at the same offset(s) as a double, it (javac) knows that the lifetimes of the bool and the int do not overlap with that of the double.  Chicory and DynComp only add one or two LocalVariables and they have a lifetime of the entire method.

Reply:
It would be helpful to document the fact that the new LocalVariables have a lifetime of the entire method, and why those definitely don't overlap with any existing variables (or why it is not a problem if they do).